src/main/java/engineering/swat/watch/Watcher.java

```diff
@@ -159,9 +168,11 @@ public ActiveWatch start() throws IOException {
             throw new IllegalStateException("There is no onEvent handler defined");
         }
 
+        var h = overflowEventHandler().andThen(userDefinedEventHandler());
```

Comment:
how about we pass the `userDefinedEventHandler` as an argument to the `overflowEventHandler`? That way for `NO_RESCANS` we can just skip the extra null handler.